apps/omg/lib/omg/state/core.ex

```diff
@@ -425,7 +424,7 @@ defmodule OMG.State.Core do
     |> Transaction.get_outputs()
     |> Enum.with_index()
     |> Enum.map(fn {output, oindex} ->
-      {Output.input_pointer(output, blknum, tx_index, oindex, tx, hash), output}
+      {Utxo.position(blknum, tx_index, oindex), output}
```

Comment:
This usage here was interesting. Since this only cares for the utxo position data, I'm not sure why we created a separate method just capture the other data and not use it. 🤔 If there's a reason we can revisit, but simplifying it to just do what it was doing underneath:

```elixir
def input_pointer(%__MODULE__{}, blknum, tx_index, oindex, _, _),
    do: Utxo.position(blknum, tx_index, oindex)
```

Reply:
To reiterate: this caters(-ed) for the output-type to input-pointer-type binding as part of https://github.com/omisego/elixir-omg/issues/844. The implementation you're putting in instead will work, it's just specific and not compatible e.g. with `output_id` input pointer, support of which used to be required (or at least impending). It also couples the `State.Core` to a specific way outputs can be referenced, so similar comments as with #1212 apply here.